src/Control/Effect/Internal/Error.hs

```diff
@@ -361,3 +403,12 @@ errorToIOSimple =
   .# introUnderMany
   .# unErrorToIOSimpleC
 {-# INLINE errorToIOSimple #-}
+
+errorToIOSimple' :: forall e m a p.
+  ( Exception e
+  , Eff (Embed IO) m
+  , MonadCatch m
+  , Threaders '[ReaderThreads] m p
+  ) => ErrorToIOSimpleC e m a
+  -> m (Either e a)
+errorToIOSimple'  = coerce errorToErrorIOSimple'
```

Comment:
I'm surprised this compiles! Figured constraints involved and type ambiguity issues would make it harder than this. Learn something new every day!

Reply:
Yeah, I was quite confused by the use of `#.` and `.#` cause `Coercible` can be lifted through any type constructors as well as `(->)`.

Reply:
I use `#.` and `.#` instead of simple `coerce`:s for a couple of reasons:

1. Some clarity as to how the stack is being reduced. A benefit to this is that each `#.` and `.#` can be replaced by `.` and it would reflect how the interpreter would be written in "normal", safe code (which doesn't depend on how `Carrier`s are represented internally, just the type signatures of the interpreters).
2. In my experience, it's _way_ too common that GHC can't figure out the types with `coerce` alone, and needs the `#.` and `.#` to be nudged in the right direction. So I made `#.` and `.#` standard use in the library since I needed them so often anyway.
3. I've made standard that an interpreter places the same constraints needed for its carrier to actually be a `Carrier`, so an interpreter advertises exactly what's needed in order for it to actually used in practice -- even if the constraints are redundant for the interpreter itself. This acts as documentation and helps out with troubleshooting.
One way I can ensure that I actually do that when I write interpreters is that I "use" the normal combinators for interpretation together with `#.` and `.#`, instead of simply having a single `coerce` or two. This is because the normal combinators require these additional constraints in order to be used, so if I forget a relevant constraint I get a compile-time error. (This is a moot point in this case since `errorToErrorIOSimple'` also requires the relevant constraints.)